crates/core/src/subscription/subscription.rs

```diff
@@ -160,41 +175,301 @@ impl QuerySet {
         let mut database_update: DatabaseUpdate = DatabaseUpdate { tables: vec![] };
         let mut seen = HashSet::new();
 
-        for query in &self.0 {
-            for q in &query.queries {
-                if let Some(t) = q.source.get_db_table() {
-                    for table in run_query(relational_db, tx, q, auth)? {
-                        {
-                            let mut table_row_operations = Vec::new();
-
-                            for row in table.data {
-                                let row_pk = pk_for_row(&row);
-
-                                //Skip rows that are already resolved in a previous subscription...
-                                if seen.contains(&(t.table_id, row_pk)) {
-                                    continue;
-                                }
-                                seen.insert((t.table_id, row_pk));
-
-                                let row_pk = row_pk.to_bytes();
-                                let row = row.data;
-                                table_row_operations.push(TableOp {
-                                    op_type: 1, // Insert
-                                    row_pk,
-                                    row,
-                                });
+        for q in &self.0 {
+            if let Some(t) = q.source.get_db_table() {
+                for table in run_query(relational_db, tx, q, auth)? {
+                    {
+                        let mut table_row_operations = Vec::new();
+
+                        for row in table.data {
+                            let row_pk = pk_for_row(&row);
+
+                            //Skip rows that are already resolved in a previous subscription...
+                            if seen.contains(&(t.table_id, row_pk)) {
```

Comment:
Ah I think this could also just check the return value of insert, so we don’t copy the pk twice.